src/ir/opt/cfold.rs

```diff
@@ -244,6 +245,35 @@ pub fn fold_cache(node: &Term, cache: &mut TermCache<Term>) -> Term {
                     b.width() + w,
                 ))))
             }),
+            Op::Map(op) => match (get(0).as_array_opt(), get(1).as_array_opt()) {
+                (Some(a), Some(b)) => {
+                    assert!(a.size == b.size);
+                    let mut new_map: BTreeMap<Value, Value> = BTreeMap::new();
+                    for (a_ind, a_val) in &a.map {
+                        let b_val = &b.map[a_ind];
+                        let res = fold_cache(
```

Comment:
@kwantam How would I evaluate the operation on `va` and `vb` without the recursive call to fold_cache here?

Reply:
High-level answer: the `Op::Map(op)` match clause starting on line 248 should push a new term onto the stack that is equivalent to the term containing the `Op::Map(_)`, but which eliminates the map. I can give you more details later.